lib/iris/cube.py

```diff
@@ -3463,29 +3491,55 @@ def slices(self, ref_to_slice, ordered=True):
             A mix of input types can also be provided. They must all be
             orthogonal (i.e. point to different dimensions).
         ordered : bool, default=True
-            If True, the order which the coords to slice or data_dims
-            are given will be the order in which they represent the data in
-            the resulting cube slices.  If False, the order will follow that of
-            the source cube.  Default is True.
+            If True, subcube dimensions are ordered to match the dimension order
+            in `ref_to_slice`. If False, the order will follow that of
+            the source cube.
 
         Returns
         -------
         An iterator of subcubes.
 
         Examples
         --------
-        For example, to get all 2d longitude/latitude subcubes from a
-        multi-dimensional cube::
+        For example, for a cube with dimensions `realization`, `time`, `latitude` and
+        `longitude`:
 
-            for sub_cube in cube.slices(['longitude', 'latitude']):
-                print(sub_cube)
+            >>> fname = iris.sample_data_path('GloSea4', 'ensemble_01[01].pp')
+            >>> cube = iris.load_cube(fname, 'surface_temperature')
+            >>> print(cube)  # doctest: SKIP
+            surface_temperature / (K)           (realization: 2; time: 6; latitude: 145; longitude: 192)
+            Dimension coordinates:
+                realization                             x        -            -               -
+                time                                    -        x            -               -
+                latitude                                -        -            x               -
+                longitude                               -        -            -               x
+            Auxiliary coordinates:
+                forecast_period                         -        x            -               -
+            Scalar coordinates:
+                forecast_reference_time     2011-07-23 00:00:00
+            Cell methods:
+                0                           time: mean (interval: 1 hour)
+            Attributes:
+                STASH                       m01s00i024
+                source                      'Data from Met Office Unified Model'
+                um_version                  '7.6'
+
+        To get all 12x2D longitude/latitude subcubes:
+
+            >>> for sub_cube in cube.slices(['longitude', 'latitude']):
+            ...     print(sub_cube)  # doctest: SKIP
+
+        To get all realizations as 2x3D separate subcubes, using dimension indices:
+
+            >>> for sub_cube in cube.slices([1, 2, 3]):
+            ...     print(sub_cube)  # doctest: SKIP
 
         See Also
         --------
         iris.cube.Cube.slices :
```

Comment:
Think this should be referencing `slices_over()` instead

Reply:
I think this is OK... could you re-check now the doctest are complete?

Reply:
The problem persists, but I think I wasn't clear about which problem I was highlighting!

From `slices_over()`:

https://github.com/SciTools/iris/blob/f06aa462d9bd7caa2c3513689b3d8a1e8d5a8fba/lib/iris/cube.py#L3446-L3449

From `slices()`:

https://github.com/SciTools/iris/blob/f06aa462d9bd7caa2c3513689b3d8a1e8d5a8fba/lib/iris/cube.py#L3541-L3544

I believe the documentation for `slices()` should be referencing `slices_over()`, not itself?

Also totally not your fault, but while we're here...

Reply:
Ah yes, I see. Fixed